src/utils/api.js

```diff
@@ -58,12 +69,64 @@ export const getOrganisationUnits = () =>
  * @param {string} nameFilter
  * @return {Promise}
  */
-export const getStandardReports = (page, pageSize, nameFilter) =>
-    api.get(createGetStandardReportsUrl(page, pageSize, nameFilter))
+export const getFilteredStandardReports = (page, pageSize, nameFilter) =>
```

Comment:
I personally find this solution using `pipe` and `lodash` overly complex. To me this is a shame because what we are actually doing is seems fairly straightforward. Please consider this alternative suggestion which I find a lot easier to follow:
```javascript
// api.js
// I've removed the getStandardReports helper, and are just calling list in the
// function body, because this helper was just a simple call to the d2 API.
// I'm calling formatStandardReportsResponse in the .then, 
// so it can just be a simple sync function
export const getFilteredStandardReports = (page, pageSize, nameFilter) =>
    modelWithFilter(d2.models.report, nameFilter)
        .list({ page, pageSize, fields: standardReportsFields })
        .then(formatStandardReportsResponse)

// helpers.js
// This now no longer needs to await the promise but can just return implicitly
const formatStandardReportsResponse = model => ({
    reports: mapResponseToJSArray(model),
    pager: {
        pageSize: model.pager.query.pageSize,
        page: model.pager.page,
        pageCount: model.pager.pageCount,
        total: model.pager.total,
    },
})

// I've written this as a generic function which can receive a model,
// I think we can re-use for a different section which has a filter above a list
const modelWithFilter = (model, filterStr) =>
    filterStr
        ? model
              .filter()
              .on('displayName')
              .ilike(filterStr)
        : model
```

Reply:
If you accept and implement this solution, please also remove lodash from the dependencies.